### PR TITLE
healpix: easyblock updated to work with latest version of healpix configure.sh

### PR DESCRIPTION
**Pull request by jose-d**
There is [new version](https://healpix.sourceforge.io/) of Healpix out including its `configure` script having even more curious questions.
These are answered here in this easyblock, and additionally, as `libpsht_healpix_f.a` is apparently not produced, let's move the sanity check to the _easyconfig_ to have better flexibility. Another good reason is possibility to use `sanity_check_commands` in the easyconfig.

### CODE REVIEW

File: easybuild/easyblocks/h/healpix.py [hunk 1]
```diff
@@ -108,15 +113,19 @@ def install_step(self):
...
+        # since 3.50 files changed, so we specify it in easyconfig.
+        # "else" section below is for backward-compatibility, ATM only easyconfig for 2.20a-icte is known
+        if LooseVersion(self.version) >= LooseVersion('3.50'):
+            super(EB_HEALPix, self).sanity_check_step()
+        else:
+            custom_paths = {
+                'files': [os.path.join('bin', x) for x in ['alteralm', 'anafast', 'hotspot', 'map2gif',
+                                                           'median_filter', 'plmgen', 'sky_ng_sim',
+                                                           'sky_ng_sim_bin', 'smoothing', 'synfast', 'ud_grade']] +
+                         [os.path.join('lib', 'lib%s.a' % x) for x in ['chealpix', 'gif', 'healpix', 'hpxgif',
```
[the remark this answers is not in the document]
Comment (jose-d) [reply easybuild/easyblocks/h/healpix.py]: this is IMO ok..  Checked with Pycharm IDE..

File: easybuild/easyblocks/h/healpix.py [hunk 2]
```diff
@@ -108,15 +113,19 @@ def install_step(self):
...
+                                                                       'psht_healpix_f']] +
+                         [os.path.join('lib', 'libchealpix.%s' % get_shared_lib_ext())],
```
[the remark this answers is not in the document]
Comment (jose-d) [reply easybuild/easyblocks/h/healpix.py]: this is IMO ok..  Checked with Pycharm IDE..

**Comment by vanzod** [comment]
I agree on moving the sanity check to the easyconfig.

**Comment by boegel** [comment]
Not a big fan of pushing the `sanity_check_paths` to the easyconfigs for recent versions, since that'll lead to duplicating them across multiple easyconfigs, which makes maintaining them harder...

If an easyblock is available, then the easyconfigs should be trimmed down where possible imho.

Note that the paths specified in the easyblock can *always* be overruled by using `sanity_check_paths` in the easyconfig file...
But if we can avoid it, i.e. if it's just a matter of only checking for some files for older versions, then we should really favor doing this in the easyblock.

As it is now, `eb` will fall back to only checking for a non-empty `bin` and `lib` directory if no custom `sanity_check_paths` is specified for recent versions, which is a lot less strict than checking for specific files.

**Comment by jose-d** [comment]
> ..

technical question: how to approach this, can the merge be "unmerged" (I doubt but not fully sure) or the fix|improvement should be created as new PR to `develop`?

**Comment by boegel** [comment]
@jose-d It should be followed up with another PR to `develop`, no need to "unmerge".

`develop` should be considered "unstable" (although it rarely is in practice), and in this particular case it's not a major problem, it's not that a bug has been introduced, strictly speaking...

Would you be able to look into that in the coming days @jose-d? I'm planning another EasyBuild release (v3.9.3) soon, I'd love to include those additional changes in there.

**Pull request merged**